homeassistant/components/garmin_connect/sensor.py

```diff
@@ -0,0 +1,255 @@
+        ): vol.All(cv.ensure_list, [vol.In(GARMIN_CONDITIONS_LIST)]),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Garmin Connect component."""
+
+    email = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+    prefix_name = config.get(CONF_NAME)
+
+    try:
+        garmin_client = garminconnect.Garmin(email, password)
+    except ValueError as err:
+        _LOGGER.error("Error occured during Garmin Connect Client init: %s", err)
+        return
+
+    garmin_data = GarminConnectClient(garmin_client)
+
+    entities = []
+    for resource in config[CONF_MONITORED_CONDITIONS]:
+        sensor_type = resource
+        name = prefix_name + " " + GARMIN_CONDITIONS_LIST[resource][0]
+        unit = GARMIN_CONDITIONS_LIST[resource][1]
+        icon = GARMIN_CONDITIONS_LIST[resource][2]
+
+        _LOGGER.debug(
+            "Registered new sensor: %s, %s, %s, %s", sensor_type, name, unit, icon
+        )
+        entities.append(GarminConnectSensor(garmin_data, sensor_type, name, unit, icon))
+
+    add_entities(entities, True)
+
+
+class GarminConnectClient:
+    """Set up the Garmin Connect client."""
+
+    def __init__(self, client):
+        """Initialize the client."""
+        self.client = client
+        self.data = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Fetch the latest data."""
+        today = date.today()
+        try:
+            self.data = self.client.fetch_stats(today.strftime("%Y-%m-%d"))
+        except ValueError as err:
+            _LOGGER.error("Error occured while fetching Garmin Connect data: %s", err)
+            return
+
+
+class GarminConnectSensor(Entity):
+    """Representation of a Garmin Connect Sensor."""
+
+    def __init__(self, data, sensor_type, name, unit, icon):
+        """Initialize the sensor."""
+        self._data = data
+        self._type = sensor_type
+        self._name = name
+        self._icon = icon
+        self._unit = unit
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def device_state_attributes(self):
+        """Return attributes for sensor."""
+
+        attributes = {}
+        if self._data.data:
+            attributes = {
+                "source": self._data.data["source"],
+                "last synced (GMT)": self._data.data["lastSyncTimestampGMT"],
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+            }
+        return attributes
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
Why throttle this, `self.-data.update` is already throttled?

Reply:
Fixed it